Modules/arraymodule.c

```diff
@@ -1692,19 +1797,24 @@ array_array_tolist_impl(arrayobject *self)
     return NULL;
 }
 
+
 static PyObject *
-frombytes(arrayobject *self, Py_buffer *buffer)
+frombytes_lock_held(arrayobject *self, PyObject *bytes)
```

Comment:
For the `_lock_held` functions, it would be nice if you could assert[^1] that the lock is actually held.

[^1]: `_Py_CRITICAL_SECTION_ASSERT_OBJECT_LOCKED`